custom-prompt/custom-prompt.cc

```diff
@@ -163,6 +166,190 @@ void Interval::print_long(std::ostream& ostream) const
  *****************************************************************************/
 extern "C" long long unsigned get_active_wid(void);
 
+/******************************************************************************
+ * Store information about the current Git repository.
+ *****************************************************************************/
+class GitRepository
+{
+private:
+    C::git_repository* repo;
+    std::filesystem::path gitdir;
+    C::git_reference* ref;
+    C::git_oid const* oid;
+    std::string description;
+    bool detached;
+    bool bisecting, cherrypicking, merging, rebasing, reverting;
+
+public:
+    GitRepository(void);
+    std::string get_information(void);
+
+private:
+    void set_description(void);
+    void set_state(void);
+    void set_status(void);
```

Comment:
#### Suggestion

Rename to `set_dirty_staged_untracked`. (Likewise, rename the callback below to `update_dirty_staged_untracked`.) Add three variables with those names.

Reply:
Done. Renamed to `establish_*` because the methods are not setters.